### PR TITLE
add Variable.mean

### PR DESCRIPTION
**Pull request by yuyu2172**
I added this method similar in spirit with `chainer.Variable.reshape`.

### CODE REVIEW

File: tests/chainer_tests/test_variable.py [hunk 1]
```diff
@@ -2359,6 +2359,61 @@ def test_chainerx(self):
...
+            w_shape = tuple(self.shape[a] for a in self.axis)
+        self.w = np.random.uniform(-2, 2, w_shape).astype(np.float32)
+        self.x = np.random.uniform(-1, 1, self.shape).astype(np.float32)
+
+    def check_forward(self, x_data, w_data):
+        x = chainer.Variable(x_data)
+        y_gt = F.mean(x, self.axis, w_data, self.keepdims)
+        y = x.mean(self.axis, w_data, self.keepdims)
+        assert y.data.dtype == y_gt.dtype
+        testing.assert_allclose(y_gt.data, y.data)
```
Comment (crcrpar) [inline tests/chainer_tests/test_variable.py]:
```suggestion
        testing.assert_allclose(y_gt.array, y.array)
```

**Comment by niboshi** [comment]
I don't think we're going to implement all the member methods in `numpy.ndarray`.
Existing ones (`shape` and `transpose`) are understandable because they're really fundamental operations of tensors.
We need some more discussion whether to make `Variable` more numpy-compatible object, and to what level.

**Comment by yuyu2172** [comment]
Do you have a reason why we shouldn't implement member methods in numpy.ndarray for Variable?
The number of member methods is finite and not too large I believe.
Although Chainer doesn't require compatibility with NumPy (unlike CuPy), the API respects NumPy a lot. Wouldn't it be helpful to make Chainer behave closer to NumPy when it is possible?

I think methods that are differentiable should be supported. For instance, 
- clip
- max
- min
- sum
- var

**Comment by niboshi** [comment]
Besides those, `numpy.ndarray` has the following differentiable methods:
* `astype`, `choose`, `compress`, `conj`, `conjugate`, `copy`, `cumprod`, `cumsum`, `diagonal`, `dot`, `flatten`, `partition`, `prod`, `ptp`, `ravel`, `repeat`, `squeeze`, `std`, `swapaxes`, `take`, `trace`, `view`, (attributes: `flat`, `imag`, `real`)



The reason why I don't think we're going to support all of those include:

* Chainer as a whole is not meant to be NumPy-compatible. If only `Variable` were made NumPy-compatible, that would make the framework inconsistent. For example, one would expect NumPy-compatible functions exist in `chainer.functions`.
* ChainerX is designed to be NumPy-compatible. It may be more productive to pour efforts to support NumPy-comptible features there.

**Comment by niboshi** [comment]
I'm not strongly insisting we should not have `Variable.mean` (I understand the motivation). I'm just saying we need a concensus about whether to make `Variable` more NumPy-compatible, and to what level.

**Comment by yuyu2172** [comment]
> ChainerX is designed to be NumPy-compatible. It may be more productive to pour efforts to support NumPy-compatible features there.

I see. I didn't know the design goal of ChainerX. Thanks.

> we need a concensus about whether to make Variable more NumPy-compatible, and to what level.
OK. I feel that it is a bit arbitrary to only allow `reshape` and `transpose` as member methods. However, as you said, extending the support to all methods takes some effort, which may be too much.

**Comment by beam2d** [comment]
> we need a concensus about whether to make Variable more NumPy-compatible, and to what level

Not for NumPy _compatibility_, but for providing similar coding experience with NumPy, I want these methods to be added to Variable. I think it is natural for Variable to provide these methods, and at least for me it is stressful to get an error when I try to use such methods.

As for the efforts needed for them, I think it's good to add them with minimal effort. For example, the current PR adds a new test case, but it would be acceptable (or even better in terms of DRY) to just add a parameter to the existing tests for `F.foobar` functions.

**Comment by emcastillo** [comment]
Please fix travis failures, also address @crcrpar comment.

**Comment by yuyu2172** [comment]
Thanks for the feedback. I will include tests to `test_average.py`.

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Travis failure is unrelated

**Pull request merged**